apps/trackers/tests/test_bugzilla.py

```diff
@@ -478,3 +478,106 @@ def test_generate_keywords(self):
+        Test that CC lists are generated for a new tracker
+        """
+
+        # For brevity of pytest.mark.parametrize's arguments
+        if component_cc:
+            component_cc = {"component": ["a@redhat.com", "a2", "ee"]}
+        else:
+            component_cc = {}
+        if private_tracker_cc:
+            private_tracker_cc = ["b@redhat.com", "b2", "ee"]
+        else:
+            private_tracker_cc = []
+        if default_cc:
+            default_cc = ["c@redhat.com", "c2", "ee"]
+        else:
+            default_cc = []
+        if exists:
+            external_system_id = "1234"
+        else:
+            external_system_id = ""
+
+        ps_module = PsModuleFactory(
+            bts_name="bugzilla",
+            component_cc=component_cc,
+            private_tracker_cc=private_tracker_cc,
+            default_cc=default_cc,
+        )
+        ps_update_stream = PsUpdateStreamFactory(
+            ps_module=ps_module,
+        )
+        flaw = FlawFactory()
+        affect = AffectFactory(
+            flaw=flaw,
+            ps_module=ps_module.name,
+            affectedness=Affect.AffectAffectedness.AFFECTED,
+            ps_component=component,
+        )
+        tracker = TrackerFactory(
+            affects=[affect],
+            type=Tracker.TrackerType.BUGZILLA,
+            ps_update_stream=ps_update_stream.name,
+            external_system_id=external_system_id,
+            embargoed=flaw.embargoed,
+        )
+
+        # create query
+        query = TrackerBugzillaQueryBuilder(tracker).query
+
+        if exists:
+            assert "cc" not in query
+        else:
+            if component == "component":
+                expected_component_cc = component_cc.get("component", [])
+            else:
+                # If the PS module's component is not listed in product definition's component_cc,
+                # there is no match.
+                expected_component_cc = []
+
+            if tracker.embargoed:
+                # BugzillaTrackerQueryBuilder uses AffectCCBuilder for creating the CC list.
+                # AffectCCBuilder doesn't implement private_tracker_cc (as of April 2024).
+                # This is not a problem because we will file BZ trackers only for public
+                # projects (probably only Fedora).
+
+                # Being explicit.
+                # NOT: expected_private_tracker_cc = private_tracker_cc
+                expected_private_tracker_cc = []
+            else:
+                expected_private_tracker_cc = []
+
+            # Unlike Jira, for BZ all names must be converted to emails
+            expected_cc = sorted(
+                set(
+                    f"{n}@redhat.com" if "@" not in n else n
+                    for n in (
+                        expected_component_cc + expected_private_tracker_cc + default_cc
+                    )
+                )
+            )
+
+            if flaw.embargoed and not ps_module.private_trackers_allowed:
```

Comment:
This is explicitly coded (but without a comment) in AffectCCBuilder, but I can't find a similar thing in SFM2 for _trackers_. Is this a flaw-only affect-only thing not applicable to trackers?

Reply:
When private trackers are not allowed but the flaw is embargoed, we should not create any trackers at all so then the tracker CC list seems irrelevant to me. I am not very sure what is the reasoning behind this functionality but it is probably some exceptional product workflow.

Reply:
Makes sense. Leaving the check in as it seems to not hurt anything if trackers are created in this case that shouldn't happen.